pkg/master/controller.go

```diff
@@ -47,6 +47,7 @@ import (
 type Controller struct {
 	NamespaceRegistry namespace.Registry
 	ServiceRegistry   service.Registry
+	ServiceRest       *service.REST
```

Comment:
@smarterclayton @lavalamp This PR is not yet fully finished, but before I fixed unit tests (which will be quite painful), I would like to get your opinion about the approach for fixing it and if you have any better suggestions how to do it.